File: WorldAround.Domain/Models/Trips/TripModel.cs
using WorldAround.Domain.Models.Comments;

namespace WorldAround.Domain.Models.Trips;

public class TripModel
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public int? AuthorId { get; set; }

    public List<PinModel> Pins { get; set; }
    public List<CommentModel> Comments { get; set; }

    public DateTime CreateDate { get; set; }
}

File: WorldAround.UI/src/app/models/trips/tripModel.ts
import { CommentModel } from "../comments/comment";
import { PinModel } from "./pin";

export class TripModel {
    id: number;
    authorId: number;
    name: string;
    description: string;
    pins: PinModel[];
    comments: CommentModel[];
    createDate: string;
}